### PR TITLE
add reuse option, add improve create logic

### PR DESCRIPTION
**Pull request by Mistobaan**
This PR tries to fix issues #26 , #32

### CODE REVIEW

File: helm/resource_release.go [hunk 1]
```diff
@@ -382,19 +442,28 @@ var all = []release.Status_Code{
...
-func getRelease(client helm.Interface, d *schema.ResourceData) (*release.Release, error) {
-	name := d.Get("name").(string)
-
+func getRelease(client helm.Interface, name string) (*release.Release, error) {
 	res, err := client.ReleaseContent(name)
 	if err != nil {
-		msg := grpc.ErrorDesc(err)
-		if strings.Contains(msg, "not found") {
-			return nil, ErrReleaseNotFound
+		if status, ok := status.FromError(err); ok {
```
Comment (mcuadros) [inline helm/resource_release.go]: why two this alternative method?

File: helm/resource_release.go [hunk 2]
```diff
@@ -321,17 +381,17 @@ func getChart(d *schema.ResourceData, m *Meta) (c *chart.Chart, path string, err
 		d.Get("keyring").(string),
 	)
 	if err != nil {
-		return
+		return c, path, err
```
Comment (mcuadros) [inline helm/resource_release.go]: not needed, since the return args are named

File: helm/resource_release.go [hunk 3]
```diff
@@ -164,20 +178,54 @@ func resourceReleaseCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
+	name := d.Get("name").(string)
+out:
+	for {
```
Comment (mcuadros) [inline helm/resource_release.go]: maybe you can move this logic to a function

**Review by mcuadros — CHANGES_REQUESTED** [review]

**Comment by mcuadros** [comment]
Nice PR, but for the next time will be better if you can send several PR or at least several commits.

**Comment by Mistobaan** [comment]
Will break it in a nicer way. This was just a quick show of the changes to comment.

File: helm/resource_release.go (hunk 3 above)
Comment (Mistobaan) [reply helm/resource_release.go]: yes. Double check if the logic makes sense. Especially if the state of the chart is unknown, should we abort, wait or try to install the new chart anyway?

**Review by mcuadros — APPROVED** [review]

**Comment by sagikazarmark** [comment]
Can we expect a release in the near future?

**Comment by sagikazarmark** [comment]
Ping @mcuadros 

This PR fixed one of the two major issues with this provider, would be nice to see a release.

**Comment by mcuadros** [comment]
@sagikazarmark https://github.com/mcuadros/terraform-provider-helm/releases/tag/v0.5.0

**Comment by sagikazarmark** [comment]
Awesome, thanks

**Pull request merged**